src/Middleware/WebSockets/src/ServerWebSocket.cs

```diff
@@ -0,0 +1,79 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net.WebSockets;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.WebSockets;
+
+/// <summary>
+/// Used in ASP.NET Core to wrap a WebSocket with its associated HttpContext so that when the WebSocket is aborted
+/// the underlying HttpContext is aborted. All other methods are delegated to the underlying WebSocket.
+/// </summary>
+internal sealed class ServerWebSocket : WebSocket
+{
+    private readonly WebSocket _wrappedSocket;
+    private readonly HttpContext _context;
+
+    internal ServerWebSocket(WebSocket wrappedSocket, HttpContext context)
+    {
+        ArgumentNullException.ThrowIfNull(wrappedSocket);
+        ArgumentNullException.ThrowIfNull(context);
+
+        _wrappedSocket = wrappedSocket;
+        _context = context;
+    }
+
+    public override WebSocketCloseStatus? CloseStatus => _wrappedSocket.CloseStatus;
+
+    public override string? CloseStatusDescription => _wrappedSocket.CloseStatusDescription;
+
+    public override WebSocketState State => _wrappedSocket.State;
+
+    public override string? SubProtocol => _wrappedSocket.SubProtocol;
+
+    public override void Abort()
+    {
+        _context.Abort();
```

Comment:
should this also call `wrappedSocket.Abort();` ? or is that reliably handled via `context.Abort()`?

Reply:
Might as well delegate the call too.